### PR TITLE
Test: simplify adding new unit tests

### PR DESCRIPTION
**Pull request by alfred2g**
Simplify Unit test cmake for a better user experience adding new unit tests

Description
-----------
Simplify the cmake file for unit tests

Checklist:
----------

- [x ] I have tested my changes. No regression in existing tests.
- [ x] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/pkcs11/utest/CMakeLists.txt [hunk 1]
```diff
@@ -1,142 +1,135 @@
...
-       "${3rdparty_dir}/mbedtls/include"
-       "${3rdparty_dir}/mbedtls_utils"
-       "${standard_dir}/crypto/include"
-       "${AFR_ROOT_DIR}/tests/include"
-       "${AFR_ROOT_DIR}/libraries/c_sdk/standard/common/include/private"
-       "${kernel_dir}/include"
-       "${CMAKE_CURRENT_BINARY_DIR}/mocks"
+# list the directories the module under test includes
+    list(APPEND real_include_directories
+                "${3rdparty_dir}/pkcs11" # should come first
```
Comment (lundinc2) [inline libraries/abstractions/pkcs11/utest/CMakeLists.txt:61]: This is getting included twice now, is that necessary?

File: libraries/freertos_plus/standard/pkcs11/utest/CMakeLists.txt [hunk 2]
```diff
@@ -1,93 +1,116 @@
...
-preprocess_mock_list(pkcs11_mock ${preprocessed_mock_list} "${unix_flags}")
-
-
-target_compile_definitions(pkcs11_mock PUBLIC
-        portHAS_STACK_OVERFLOW_CHECKING=1
-        portUSING_MPU_WRAPPERS=1
-        MPU_WRAPPERS_INCLUDED_FROM_API_FILE
+    project ("freertos_plus pkcs11 cmock unit test")
+    cmake_minimum_required (VERSION 3.13)
+    set(3rdparty_dir "${AFR_ROOT_DIR}/libraries/3rdparty")
```
Comment (lundinc2) [inline libraries/freertos_plus/standard/pkcs11/utest/CMakeLists.txt:3]: looks like we can remove these to align with the other mock suites

Comment (alfred2g) [reply libraries/freertos_plus/standard/pkcs11/utest/CMakeLists.txt:3]: oups, i forgot about this one
i will remove it

File: tools/cmock/create_test.cmake [hunk 3]
```diff
@@ -109,10 +122,42 @@ function(create_mock_list mock_name mock_list cmock_config)
...
+                             real_include_list
+                             mock_name)
+    add_library(${target} STATIC
+            ${src_file}
+        )
+    target_include_directories(${target} PUBLIC
+            ${real_include_list}
+            )
+    set_target_properties(${target} PROPERTIES
+                COMPILE_FLAGS " \
```
Comment (lundinc2) [inline tools/cmock/create_test.cmake:149]: It might be desirable to also add -Wpedantic and -Wextra for the real code. What are your thoughts on this?

Comment (alfred2g) [reply tools/cmock/create_test.cmake:149]: sounds good, i will add them, there is a failing check I will fix both

File: vendors/pc/boards/linux/CMakeLists.txt [hunk 4]
```diff
@@ -2,10 +2,18 @@ if (AFR_ENABLE_UNIT_TESTS)
 
     set(CMAKE_C_FLAGS_DEBUG "-ggdb3 -Og")
     set(CMAKE_C_FLAGS_RELEASE "-O3")
-    set(CMAKE_C_FLAGS "-Wall")
+    set(CMAKE_C_FLAGS "-Wall -Wextra -Wpedantic")
```
Comment (lundinc2) [inline vendors/pc/boards/linux/CMakeLists.txt:5]: Does this add these warnings to the mocks as well? That might create a lot of noise when compiling

Comment (alfred2g) [reply vendors/pc/boards/linux/CMakeLists.txt:5]: Yes it does actually, I will fix that

File: vendors/pc/boards/linux/CMakeLists.txt [hunk 5]
```diff
@@ -6,6 +6,14 @@ if (AFR_ENABLE_UNIT_TESTS)
 
     set(CMAKE_SHARED_LINKER_FLAGS "-fPIC")
 
+    set(3rdparty_dir "${AFR_ROOT_DIR}/libraries/3rdparty")
```
Comment (yourslab) [inline vendors/pc/boards/linux/CMakeLists.txt:9]: We could also use AFR_3RDPARTY_DIR, right? Then there's AFR_MODULES_*_DIR. Any reason why we're not using these variables declared in afr.cmake instead?

**Pull request merged**